var/spack/repos/builtin/packages/py-lightning/package.py

```diff
@@ -16,16 +16,13 @@ class PyLightning(PythonPackage):
 
     maintainers("adamjstewart")
 
+    version("2.0.4", sha256="f5f5ed75a657caa8931051590ed000d46bf1b8311ae89bb17a961c3f299dbf33")
     version("2.0.3", sha256="5a70f05e40f1d7882f81eace0d4a86fe2604b423f8df42beaabd187bfdb420cf")
     version("2.0.2", sha256="fa32d671850a5be2d961c6705c927f6f48d1cf9696f61f7d865244142e684430")
     version("2.0.1", sha256="abf4f9e10b0d97348336038db79f4efc75daa2f3f81876822273023294d6ef3e")
     version("2.0.0", sha256="dfe158aa91ac139d8bdfccc7cdb627072e0052076ae9c0459c8fa12a028dbe6c")
     version("1.9.5", sha256="4a6ee1bf338f7677f04d339b84dd0c9c0fa407c3dacea366a111dc86476d4dec")
 
-    variant(
```

Comment:
Would it make sense to, instead of removing it complete, only make it available for older releases (similarly to how I have done it in `py-mne+full`). Just in case if anyone other than you uses it?

Reply:
@alalazo this is why I want a feature to deprecate variants.

Reply:
What would you need more than conditional variant? A way to express that if the variant is selected a warning message should be issued?

Reply:
Basically, yeah. Like I could make it conditional on old versions but then I would have to wait until all those versions are deprecated before removing it. Maybe a warning or concretization penalty when selecting a value different from the default.